src/object.c

```diff
@@ -424,12 +417,390 @@ int paint_path_entry(int flags, int ebx, int ecx, int edx, rct_drawpixelinfo* dp
 		{
 			if ((flags & 0xFF) <= 1)//0
 			{
-				uint8_t* ebp = esi + 0xE;
-				((rct_string_id*)esi)[0] = object_get_localised_text(&ebp, ecx, ebx, 0);
+				uint8_t* ebp = (uint8_t*)(esi + 0x1C2);
```

Comment:
Since we know esi is of the structure rct_ride_type can you change all of the below code to use that struct instead of +0x1AE and such. Any missing fields add them to the rct_ride_type struct as var_XXX.